subsys/net/openthread/rpc/client/ot_rpc_if_client.c

```diff
@@ -166,11 +168,26 @@ static int ot_rpc_l2_enable(struct net_if *iface, bool state)
 {
 	const size_t cbor_buffer_size = 1;
 	struct nrf_rpc_cbor_ctx ctx;
+	struct zcbor_string zst;
+	int error = 0;
 
 	if (!state) {
 		otRemoveStateChangeCallback(NULL, ot_state_changed_handler, iface);
 	}
 
+	net_if_set_link_addr(iface, mac, 8, NET_LINK_IEEE802154);
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, 0);
+	nrf_rpc_cbor_cmd_rsp(&ot_group, OT_RPC_CMD_IF_EXTADDR, &ctx);
+
+	if (!zcbor_bstr_decode(ctx.zs, &zst)) {
+		error = -1;
+		goto exit;
+	}
+
+	memcpy(mac, zst.value, zst.len);
```

Comment:
```suggestion
	memcpy(mac, zst.value, sizeof(mac) < zst.len ? sizeof(mac) : zst.len);
```
Otherwise, memory corruption might happen if received value is longer than mac size.

Reply:
fixed.